### ISSUE TITLE
Error uploading to Endpoint Manager/Intune

### ISSUE THREAD

**Issue by Protinon** [adoptium/adoptium-support#594]
### Please provide a brief summary of the bug

Our company uses Microsoft Endpoint Manager for pushing software to our clients' computers. EM allows for direct msi uploads (much easier than .exe installers). There is an error when uploading the installers: **The selected app package does not appear to have either a ProductCode or ProductVersion.**

This occurs with:
- OpenJDK8U-jdk_x64_windows_hotspot_8u345b01.msi
- OpenJDK19U-jdk_x64_windows_hotspot_19_36.msi

I've checked the msi files in Orca, ProductVersion and ProductCode do exist. I can test with EM on my end, and I have a support ticket open with Microsoft. Microsoft has consistently told me to contact you for assistance, even though the issue seems to be on their end.

### Expected Results

The software should be uploaded to Endpoint Manager

### Actual Results

![image](https://user-images.githubusercontent.com/38368048/193324292-1c0dcdcb-b20a-47a6-87fd-fefebe440493.png)

### What Java Version are you using?

N/A

### Did it work before?

```shell
No
```

**Comment by Protinon** [comment]
I've just copied your support email on my ticket with MS support. It's case#32718855 on their system.

**Comment by karianna** [comment]
@Protinon To do double check - what are the various versions of Windows that this package is being pushed to?  We understand that a 32-bit installer worked and I'm wondering if our 64-bit installer was being accidentally pushed to a 32-bit machine.

**Comment by Protinon** [comment]
The issue is not with pushing software to computers, but uploading them to Endpoint Manager. They install on all machines perfectly fine, but I have no way of mass-pushing Adoptium to client computers if the software will not upload to EM.

**Comment by gdams** [comment]
Inspecting the MSI more closely we do seem to have the ProductCode set so not sure why else this would fail:

```output

ProductCode = {F4A0EF62-164A-4E05-AC6F-2AE03D8685B7}
Language = 1033
ProductName = Eclipse Temurin JRE with Hotspot 20+18 (arm64)
PackageCode = {77C118E6-B6F6-4717-B298-FB82B8C4C77E}
AssignmentType = 1
PackageName = OpenJDK-jre_aarch64_windows_hotspot_2022-10-12-03-30.msi
InstalledProductName = Eclipse Temurin JRE with Hotspot 20+18 (arm64)
VersionString = 20.0.0.18
ProductIcon = C:\windows\Installer\{F4A0EF62-164A-4E05-AC6F-2AE03D8685B7}\logo.ico
InstallLocation = C:\Program Files (Arm)\Eclipse Adoptium\jre-20.0.0.18-hotspot\
InstallSource = C:\Users\georg\Downloads\
InstallDate = 20221101
Publisher = Eclipse Adoptium
LocalPackage = C:\windows\Installer\2cc11928.msi
HelpLink = https://github.com/adoptium/adoptium-support/issues/new/choose
URLInfoAbout = https://adoptium.net/support.html
URLUpdateInfo = https://adoptium.net/releases.html
```

**Comment by Protinon** [comment]
I have to agree with you, the error message shown isnt helpful in the slightest. It may not even be the actual issue.

I've been wrestling with MS support for a month now, but they are convinced the issue lies with Adoptium.

I was hoping by looping you in, they would realize the issue and maybe ask you try some possible solutions. I heard you joined a meeting with MS support, did anything come from that?

**Comment by Protinon** [comment]
> @Protinon I assume the End Point Manager is running on a 64-bit host or are you using an Azure hosted End Point Manager?

Forgot to respond to this, we are using Endpoint Manager hosted on Microsoft servers. We've never had issues uploading 32/64bit programs before.

**Comment by karianna** [comment]
In internal MSFT testing it does seem to be an Adoptium installer only issue, we're diffing that and other installers (from say Azul Zulu or Amazon Corretto).

**Comment by Tonrab** [comment]
has there been any progress on this at all?  Trying to deploy to 120+ users consumes a lot of time
